client/landing/stepper/router/index.tsx

```diff
@@ -0,0 +1,138 @@
+import React, { Children, PropsWithChildren, ReactElement, ReactNode } from 'react';
+import { Route, Routes } from 'react-router';
+import { Primitive } from 'utility-types';
+
+{
+	/* <Flow persistParam={['from', 'to', 'siteSlug']}>
+	<From step={step} >
+		<To step={step} when={hasSite} />
+		<To step={step} />
+		<BackTo step={step} when={hasSite} />
+	</From>
+
+	<From step={step} >
+		<To step={step} when={hasSite} />
+	</From>
+</Flow> */
+}
+
+const RouteWithCondition = ( {
+	children,
+	condition,
+}: {
+	children: React.ReactNode;
+	condition: () => boolean;
+} ) => {
+	return <div>{ children }</div>;
+};
+
+interface FromProps extends React.PropsWithChildren {
+	step: string;
+}
+
+interface ToProps extends React.PropsWithChildren {
+	step: string;
+	when?: () => boolean;
+}
+
+interface FlowProps {
+	name: string;
+	persistParam: string[];
+	children: React.ReactElement< FromProps >[];
+}
+
+type Step = ReactElement< FromProps > | ReactElement< ToProps >;
+const extractSteps = ( children: Step[] ) => {
+	return Children.map( children, ( child: Step ) => {
+		return child.props.step;
+	} );
+};
+
+export const Flow = ( {
+	children,
+	name,
+	persistParam,
+}: {
+	children: Step | Step[];
+	persistParam: string[];
+	name: string;
+} ) => {
+	return <Routes>{ children }</Routes>;
+};
+
+interface INavigation {
+	from: string;
+	toOptions: {
+		step: string;
+		when: ( context: IContext ) => boolean;
+	}[];
+}
+
+export type IContext< P extends Record< string, Primitive > = Record< string, Primitive > > = P;
+
+export const From = ( { children, step }: { children: React.ReactNode; step: string } ) => {
+	return <div>From</div>;
+};
+
+export const To = ( { step, when }: { step: string; when?: ( context: IContext ) => boolean } ) => {
+	return <div>To</div>;
+};
+
+export const BackTo = ( {
+	step,
+	when,
+}: {
+	step: string;
+	when: ( context: IContext ) => boolean;
+} ) => {
+	return <div>BackTo</div>;
+};
+
+const extractNavigation = ( children: ReactElement< FromProps >[] ) => {
+	return Children.map< INavigation, ReactElement< FromProps > >(
```

Comment:
I wonder if it would be better to use an object instead of React components for the flow structure.

I think we're not extracting any benefit from React components, right? So I think having an object (with its format assured by a TypeScript) could make more sense, and this logic would be easier to read/maintain in the future. WDYT?

Reply:
I agree we can double-check it. My intent here is to enable us to use hooks, especially on the when functions.